Entity/Context/DocumentImportContext.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+
+namespace EMS\CoreBundle\Entity\Context;
+
+
+use Doctrine\ORM\EntityManager;
+use EMS\CoreBundle\Entity\ContentType;
+use EMS\CoreBundle\Entity\Environment;
+use EMS\CoreBundle\Repository\ContentTypeRepository;
+use EMS\CoreBundle\Repository\RevisionRepository;
+
+class DocumentImportContext
+{
+
+    /** @var ContentType */
+    private $contentType;
+    /** @var string */
+    private $contentTypeName;
+    /** @var string */
+    private $lockUser;
+    /** @var bool */
+    private $rawImport;
+    /** @var EntityManager */
+    private $entityManager;
+    /** @var RevisionRepository */
+    private $revisionRepository;
+    /** @var bool */
+    private $indexInDefaultEnv;
+    /** @var bool */
+    private $signData;
+    /** @var ContentTypeRepository */
+    private $contentTypeRepository;
+    /** @var Environment */
+    private $environment;
+    /** @var bool */
+    private $finalize;
+    /** @var bool */
+    private $force;
+
+    public function __construct(EntityManager $entityManager, string $contentTypeName, string $lockUser, bool $rawImport, bool $signData, bool $indexInDefaultEnv, bool $finalize, bool $force)
+    {
+        $this->contentTypeName = $contentTypeName;
+        $this->indexInDefaultEnv = $indexInDefaultEnv;
+        $this->signData = $signData;
+        $this->lockUser = $lockUser;
+        $this->rawImport = $rawImport;
+        $this->finalize = $finalize;
+        $this->entityManager = $entityManager;
+        $this->force = $force;
+
+        $this->entityManager->getConnection()->getConfiguration()->setSQLLogger(null);
```

Comment:
The context is a object storage for input data, that can be accessed using getters. It should not configure anything itself. It should be used by the DocumentService to correctly configure it's dependencies on every function call that is made.

Reply:
5fc390743659350eed5c9f5595127075aa3dda21 14d05b91a40b4052bb22a4ed7e59a025063362ab